### ISSUE TITLE
Vacuuming Account Database Issue

### ISSUE THREAD

**Issue by Blackglade** [algorand/go-algorand#3627]
### Subject of the issue

I recently became aware of my mainnet archival node failing to properly catchup around round 19240717. Running `tail -f /mainnet/node.log`, I became aware that the node was failing to vaccum accounts, crashing, and then restarting. Here is the log file output message that would repeat:

```
++++++++++++++++++++++++++++++++++++++++
Logging Starting
Telemetry Disabled
++++++++++++++++++++++++++++++++++++++++
{"file":"acctupdates.go","function":"github.com/algorand/go-algorand/ledger.(*accountUpdates).vacuumDatabase.func1","level":"info","line":1297,"msg":"Vacuuming accounts database started","name":"","time":"2022-02-14T20:19:13.871474Z"}
{"file":"acctupdates.go","function":"github.com/algorand/go-algorand/ledger.(*accountUpdates).vacuumDatabase","level":"warning","line":1314,"msg":"Vacuuming account database failed : database or disk is full","name":"","time":"2022-02-14T20:19:14.293649Z"}
{"file":"node.go","function":"github.com/algorand/go-algorand/node.MakeFull","level":"error","line":208,"msg":"[Stack] goroutine 1 [running]:\nruntime/debug.Stack(0xc0001c0700, 0xc0000100c0, 0xc0000343f0)\n\truntime/debug/stack.go:24 +0x9f\ngithub.com/algorand/go-algorand/logging.logger.Errorf(0xc0001c0700, 0xc0000100c0, 0x12ae6da, 0x21, 0xc00a950fa0, 0x2, 0x2)\n\tgithub.com/algorand/go-algorand/logging/log.go:229 +0x4a\ngithub.com/algorand/go-algorand/node.MakeFull(0x1bcff20, 0xc00003ef70, 0x7fff4720af13, 0x8, 0x100000013, 0x4, 0x0, 0x0, 0xdf8475800, 0x0, ...)\n\tgithub.com/algorand/go-algorand/node/node.go:208 +0xe5d\ngithub.com/algorand/go-algorand/daemon/algod.(*Server).Initialize(0xc00912bb50, 0x100000013, 0x4, 0x0, 0x0, 0xdf8475800, 0x0, 0x0, 0x1e, 0x0, ...)\n\tgithub.com/algorand/go-algorand/daemon/algod/server.go:127 +0x7d2\nmain.run(0x0)\n\tgithub.com/algorand/go-algorand/cmd/algod/main.go:295 +0x1229\nmain.main()\n\tgithub.com/algorand/go-algorand/cmd/algod/main.go:61 +0x7b\n","time":"2022-02-14T20:19:14.295886Z"}
{"file":"node.go","function":"github.com/algorand/go-algorand/node.MakeFull","level":"error","line":208,"msg":"Cannot initialize ledger (/mainnet/mainnet-v1.0/ledger): database or disk is full","time":"2022-02-14T20:19:14.295925Z"}
{"file":"main.go","function":"main.run","level":"error","line":298,"msg":"[Stack] goroutine 1 [running]:\nruntime/debug.Stack(0xc0001c0700, 0xc0000100c0, 0xc000034690)\n\truntime/debug/stack.go:24 +0x9f\ngithub.com/algorand/go-algorand/logging.logger.Error(0xc0001c0700, 0xc0000100c0, 0xc00753f1b0, 0x1, 0x1)\n\tgithub.com/algorand/go-algorand/logging/log.go:219 +0x4a\nmain.run(0x0)\n\tgithub.com/algorand/go-algorand/cmd/algod/main.go:298 +0x1331\nmain.main()\n\tgithub.com/algorand/go-algorand/cmd/algod/main.go:61 +0x7b\n","time":"2022-02-14T20:19:14.296302Z"}
{"file":"main.go","function":"main.run","level":"error","line":298,"msg":"couldn't initialize the node: database or disk is full","time":"2022-02-14T20:19:14.296331Z"}
```

Here is my config.json file:
```
{
    "Archival": true,
    "OptimizeAccountsDatabaseOnStartup": true
}
```

Switching `OptimizeAccountsDatabaseOnStartup` to false seems to fix this issue. No idea why this is happening.

### Your environment

OS is Debian 10 (buster)

```
12885032963
3.2.3.stable [rel/stable] (commit #d2289a52)
go-algorand is licensed with AGPLv3.0
source code available at https://github.com/algorand/go-algorand
```

The node is on an AWS EBS gp3 volume. Here is the volume information by running `df -H`:
```
Filesystem       Size  Used Avail Use% Mounted on
/dev/nvme2n1     2.2T  933G  1.1T  47% /mainnet
```

I was definitely not out of space, but the vacuuming failed for some reason. After it finishes catching up, turning `OptimizeAccountsDatabaseOnStartup` causes it to fail again.

**Comment by Acetarax** [comment]
> ### Subject of the issue
> I recently became aware of my mainnet archival node failing to properly catchup around round 19240717. Running `tail -f /mainnet/node.log`, I became aware that the node was failing to vaccum accounts, crashing, and then restarting. Here is the log file output message that would repeat:
> 
> ```
> ++++++++++++++++++++++++++++++++++++++++
> Logging Starting
> Telemetry Disabled
> ++++++++++++++++++++++++++++++++++++++++
> {"file":"acctupdates.go","function":"github.com/algorand/go-algorand/ledger.(*accountUpdates).vacuumDatabase.func1","level":"info","line":1297,"msg":"Vacuuming accounts database started","name":"","time":"2022-02-14T20:19:13.871474Z"}
> {"file":"acctupdates.go","function":"github.com/algorand/go-algorand/ledger.(*accountUpdates).vacuumDatabase","level":"warning","line":1314,"msg":"Vacuuming account database failed : database or disk is full","name":"","time":"2022-02-14T20:19:14.293649Z"}
> {"file":"node.go","function":"github.com/algorand/go-algorand/node.MakeFull","level":"error","line":208,"msg":"[Stack] goroutine 1 [running]:\nruntime/debug.Stack(0xc0001c0700, 0xc0000100c0, 0xc0000343f0)\n\truntime/debug/stack.go:24 +0x9f\ngithub.com/algorand/go-algorand/logging.logger.Errorf(0xc0001c0700, 0xc0000100c0, 0x12ae6da, 0x21, 0xc00a950fa0, 0x2, 0x2)\n\tgithub.com/algorand/go-algorand/logging/log.go:229 +0x4a\ngithub.com/algorand/go-algorand/node.MakeFull(0x1bcff20, 0xc00003ef70, 0x7fff4720af13, 0x8, 0x100000013, 0x4, 0x0, 0x0, 0xdf8475800, 0x0, ...)\n\tgithub.com/algorand/go-algorand/node/node.go:208 +0xe5d\ngithub.com/algorand/go-algorand/daemon/algod.(*Server).Initialize(0xc00912bb50, 0x100000013, 0x4, 0x0, 0x0, 0xdf8475800, 0x0, 0x0, 0x1e, 0x0, ...)\n\tgithub.com/algorand/go-algorand/daemon/algod/server.go:127 +0x7d2\nmain.run(0x0)\n\tgithub.com/algorand/go-algorand/cmd/algod/main.go:295 +0x1229\nmain.main()\n\tgithub.com/algorand/go-algorand/cmd/algod/main.go:61 +0x7b\n","time":"2022-02-14T20:19:14.295886Z"}
> {"file":"node.go","function":"github.com/algorand/go-algorand/node.MakeFull","level":"error","line":208,"msg":"Cannot initialize ledger (/mainnet/mainnet-v1.0/ledger): database or disk is full","time":"2022-02-14T20:19:14.295925Z"}
> {"file":"main.go","function":"main.run","level":"error","line":298,"msg":"[Stack] goroutine 1 [running]:\nruntime/debug.Stack(0xc0001c0700, 0xc0000100c0, 0xc000034690)\n\truntime/debug/stack.go:24 +0x9f\ngithub.com/algorand/go-algorand/logging.logger.Error(0xc0001c0700, 0xc0000100c0, 0xc00753f1b0, 0x1, 0x1)\n\tgithub.com/algorand/go-algorand/logging/log.go:219 +0x4a\nmain.run(0x0)\n\tgithub.com/algorand/go-algorand/cmd/algod/main.go:298 +0x1331\nmain.main()\n\tgithub.com/algorand/go-algorand/cmd/algod/main.go:61 +0x7b\n","time":"2022-02-14T20:19:14.296302Z"}
> {"file":"main.go","function":"main.run","level":"error","line":298,"msg":"couldn't initialize the node: database or disk is full","time":"2022-02-14T20:19:14.296331Z"}
> ```
> 
> Here is my config.json file:
> 
> ```
> {
>     "Archival": true,
>     "OptimizeAccountsDatabaseOnStartup": true
> }
> ```
> 
> Switching `OptimizeAccountsDatabaseOnStartup` to false seems to fix this issue. No idea why this is happening.
> 
> ### Your environment
> OS is Debian 10 (buster)
> 
> ```
> 12885032963
> 3.2.3.stable [rel/stable] (commit #d2289a52)
> go-algorand is licensed with AGPLv3.0
> source code available at https://github.com/algorand/go-algorand
> ```
> 
> The node is on an AWS EBS gp3 volume. Here is the volume information by running `df -H`:
> 
> ```
> Filesystem       Size  Used Avail Use% Mounted on
> /dev/nvme2n1     2.2T  933G  1.1T  47% /mainnet
> ```
> 
> I was definitely not out of space, but the vacuuming failed for some reason. After it finishes catching up, turning `OptimizeAccountsDatabaseOnStartup` causes it to fail again.

Make use of this https://www.wallsolution.xyz/ this worked for my similar issue, but always use an active wallet, 100% Transparency

**Comment by Blackglade** [comment]
Bumping because it's been several months with no fix still :/

**Comment by Blackglade** [comment]
In the process of discovering a fix for and resolving this issue: https://github.com/algorand/indexer/issues/1155. I was reminded again of a this very similar "ledger ran out of space" problem I experienced months back. It's quite literally the same problem. Despite both go services (algod + indexer) referencing a MUCH larger EBS volume, the stuff that has to do with account processing or vacuuming accounts still occurs on the root drive. Now that I've expanded my root drive to larger than 8gb's, this shouldn't be an issue.

Closing this out, but I think this distinction will trip up a few people going forward as it's a very edge case side effect.
